velox/tool/trace/tests/PartitionedOutputReplayerTest.cpp

```diff
@@ -0,0 +1,184 @@
+            makeFlatVector<std::string>(
+                80'000, [](auto row) { return std::string(1024, 'x'); })})}};
+
+  for (auto& testParam : testParams) {
+    SCOPED_TRACE(testParam.debugString());
+    std::string planNodeId;
+    auto plan =
+        PlanBuilder()
+            .values({testParam.input}, false)
+            .partitionedOutput(
+                {"key"}, testParam.numPartitions, false, {"key", "value"})
+            .capturePlanNodeId(planNodeId)
+            .planNode();
+    const auto testDir = TempDirectoryPath::create();
+    const auto traceRoot =
+        fmt::format("{}/{}", testDir->getPath(), "traceRoot");
+    auto originalTask = Task::create(
+        fmt::format(
+            "local://test-partitioned-output-replayer-basic-{}",
+            testParam.testName),
+        core::PlanFragment{plan},
+        0,
+        createQueryContext(
+            {{core::QueryConfig::kQueryTraceEnabled, "true"},
+             {core::QueryConfig::kQueryTraceDir, traceRoot},
+             {core::QueryConfig::kQueryTraceMaxBytes,
+              std::to_string(100UL << 30)},
+             {core::QueryConfig::kQueryTraceTaskRegExp, ".*"},
+             {core::QueryConfig::kQueryTraceNodeIds, planNodeId},
+             {core::QueryConfig::kMaxPartitionedOutputBufferSize,
+              std::to_string(8UL << 20)},
+             {core::QueryConfig::kMaxOutputBufferSize,
+              std::to_string(8UL << 20)}}),
+        Task::ExecutionMode::kParallel);
+    originalTask->start(1);
+
+    std::vector<std::vector<std::unique_ptr<folly::IOBuf>>>
+        originalPartitionedResults;
+    originalPartitionedResults.reserve(testParam.numPartitions);
+    originalPartitionedResults = getAllData(
+        bufferManager_,
+        originalTask->taskId(),
+        testParam.numPartitions,
+        executor_.get());
+
+    SCOPED_TESTVALUE_SET(
+        "facebook::velox::tool::PartitionedOutputReplayer::run",
+        std::function<void(void*)>([&](void* results) {
+          std::vector<std::vector<std::unique_ptr<folly::IOBuf>>>*
+              partitionedResults = reinterpret_cast<
+                  std::vector<std::vector<std::unique_ptr<folly::IOBuf>>>*>(
+                  results);
+          ASSERT_EQ(partitionedResults->size(), testParam.numPartitions);
+          for (uint32_t partition = 0; partition < testParam.numPartitions;
+               partition++) {
+            const auto& originalBufList =
+                originalPartitionedResults.at(partition);
+            const auto& replayedBufList = partitionedResults->at(partition);
+            ASSERT_EQ(replayedBufList.size(), originalBufList.size());
+            for (uint32_t i = 0; i < replayedBufList.size(); i++) {
+              ASSERT_EQ(
+                  replayedBufList[i]->computeChainDataLength(),
+                  originalBufList[i]->computeChainDataLength());
+            }
+          }
+        }));
+
+    PartitionedOutputReplayer(
```

Comment:
Can you provide a consumer call to PartitionedOutputReplayer instead of testvalue for this testing purpose? The consumer callback takes a partition and serialized page for each invocation?